client/src/main/java/org/apache/nemo/client/JobLauncher.java

```diff
@@ -441,7 +446,8 @@ private static Configuration getDeployModeConf(final Configuration jobConf) thro
           .build();
       case "yarn":
         return YarnClientConfiguration.CONF
-          .set(YarnClientConfiguration.JVM_HEAP_SLACK, injector.getNamedInstance(JobConf.JVMHeapSlack.class))
+          .set(YarnClientConfiguration.JVM_HEAP_SLACK, injector.getNamedInstance(JobConf.JVMHeapSlack.class)
+            + injector.getNamedInstance(JobConf.MaxOffheapRatio.class))
```

Comment:
Could you add the detailed comments on why the `JVM_HEAP_SLACK` should be the sum of `JVMHeapSlack + MaxOffheapRatio`?